app/Http/Controllers/PublisherController.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Resources\PublisherResource;
+use App\Models\Publisher;
+use Exception;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Validation\Rule;
+
+class PublisherController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     */
+    public function index(Request $request)
+    {
+        try {
+            $perPageOptions = [20, 50, 100];
+
+            $searchValue = $request->input('search', '');
+            $perPage = $request->input('per_page', 20); // Default to 20 if not defined
+
+            if (!in_array($perPage, $perPageOptions)) {
+                $perPage = 20;
+            }
+
+            $validator = Validator::make($request->all(), [
+                'search' => 'nullable|string',
+            ]);
+
+            if ($validator->fails()) {
+                return response()->json([
+                    'error' => 'Invalid parameters',
+                    'message' => $validator->errors()
+                ], 400);
+            }
+
+            $query = Publisher::query();
+            if (!empty($searchValue)) {
+                $query->where('name', 'LIKE', "%{$searchValue}%");
+            }
+
+            $publishers = $query->paginate($perPage);
```

Comment:
@damir-tahirovic You can write this as a single query using conditional clauses. Check this out https://laravel.com/docs/11.x/queries#conditional-clauses